NorthstarDLL/rpakfilesystem.cpp

```diff
@@ -283,6 +283,48 @@ void* ReadFullFileFromDiskHook(const char* requestedPath, void* a2)
 			requestedPath = allocatedNewPath;
 		}
 	}
+	else if (path.extension() == ".starpak")
+	{
+		// code for this is mostly stolen from above
+
+		// game adds r2\ to every path, so assume that a starpak path that begins with r2\paks\ is a vanilla one
+		// modded starpaks will be in the mod's paks folder
+		if (path.string().find("r2\\paks\\") != 0 && path.string().find("r2/paks/") != 0)
+		{
+			// remove the r2\ from the start used for path lookups
+			std::string starpakPath = path.string().substr(3);
+			size_t hashed = STR_HASH(starpakPath);
+
+			for (Mod& mod : g_ModManager->m_loadedMods)
+			{
+				if (!mod.Enabled)
+					continue;
+
+				bool found = false;
+
+				for (size_t hash : mod.StarpakPaths)
+				{
+					if (hash == hashed)
+					{
+						starpakPath = mod.ModDirectory.string() + "\\paks\\" + starpakPath;
+						allocatedNewPath = new char[starpakPath.size() + 1];
+						strncpy(allocatedNewPath, starpakPath.c_str(), starpakPath.size());
```

Comment:
strncpy_s

Reply:
I would actually simply make allocatedNewPath a c++ string, then call c_str when passing it to stuff within the function.